### ISSUE TITLE
error when run preprocess_dannce.m

### ISSUE THREAD

**Issue by yuan0821** [jessedmarshall/CAPTURE_demo#3]
Hi, Thanks for providing such an amazing method for motion capture. I met a problem that is still not resolved. Could you please check the original code and give me some suggestions for solving such an error? Thanks a lot!!!!

> median filtering getting velocities 
> Error using filtfilt
> Expected input to be finite.
> 
> Error in filtfilt>efiltfilt (line 114)
>     validateattributes(x,{'double'},{'finite','nonempty'},'filtfilt');
> 
> Error in filtfilt (line 89)
>         y=efiltfilt(b,a,x);
> 
> Error in compute_preprocessed_mocapstruct (line 45)
>     abs_velocity_antialiased(ll,:) =  filtfilt(f1,f2, marker_velocity(ll,:,4));
> 
> Error in preprocess_ratception_struct_demo (line 43)
> ratception_struct_temppreproc = compute_preprocessed_mocapstruct(ratception_struct_temp,preprocessing_parameters);

![image](https://user-images.githubusercontent.com/50692530/112748021-aef04300-8feb-11eb-8cee-244218a3edf5.png)

**Comment by yuan0821** [comment]
I wonder it is the problem during the filtfilt.?? there are actually infinite values in marker_velocity. So maybe do the interpolation for missing data first??? Please give your comments. Thanks!!!

![image](https://user-images.githubusercontent.com/50692530/112748078-10b0ad00-8fec-11eb-8bf8-703a97e4c4c8.png)

**Comment by jessedmarshall** [comment]
Yes it seems like the velocity is not well conditioned, ie has infinite values. You might just try running a check for infinite/nan values ie 
maxvelocity = 10;%AU
velocity(isinf(velocity)) = maxvelocity;

**Comment by jessedmarshall** [comment]
Is this happening in the demo code?

**Comment by yuan0821** [comment]
Thanks, Jesse. 
Yes, the velocity has nan values as the original coordinate has missing data (65 missing values for x/y/z axis).

![image](https://user-images.githubusercontent.com/50692530/112925140-f25dc500-9143-11eb-804e-d1495a0e5c2c.png)
![image](https://user-images.githubusercontent.com/50692530/112925886-30a7b400-9145-11eb-96a8-ad4eeff2dd5d.png)

**Comment by yuan0821** [comment]
I run the script that you uploaded there. The DANNCE data was used to perform a coding test (preprocess_dannce.mat). But the script does not work for the data.
